examples/Scikit.ipynb

```diff
@@ -0,0 +1,330 @@
+    "\n",
+    "# Define model, lower C value gives higher regulation\n",
+    "clf = LogisticRegression(\n",
+    "    C=0.001, penalty='l2', solver='saga', tol=0.1\n",
+    ")\n",
+    "\n",
+    "# Fit model\n",
+    "clf.fit(X_train, y_train)\n",
+    "\n",
+    "# Score model\n",
+    "score = clf.score(X_test, y_test)\n",
+    "print(\"Test score with L2 penalty: %.4f\" % score)\n",
+    "\n",
+    "# Visualize coefficients\n",
+    "coef = clf.coef_.copy()\n",
+    "max_val = np.abs(coef).max()\n",
+    "\n",
+    "plt.figure(figsize=(10, 5))\n",
+    "for i in range(10):\n",
+    "    p = plt.subplot(2, 5, i + 1)\n",
+    "    p.imshow(coef[i].reshape(28, 28), cmap=plt.cm.RdBu,\n",
+    "             vmin=-max_val, vmax=max_val)\n",
+    "    p.set_xticks(())\n",
+    "    p.set_yticks(())\n",
+    "    p.set_xlabel('Class %i' % i)\n",
+    "plt.suptitle('Classification vector for...')\n",
+    "\n",
+    "run_time = time.time() - t0\n",
+    "print('Example run in %.3f s' % run_time)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52a034e3",
+   "metadata": {},
+   "source": [
+    "## The \"Application\"\n",
+    "\n",
+    "We can use the `xaitk-saliency` high-level API to generate saliency maps that compare to the visualization above.\n",
+    "\n",
+    "Our \"application\" will accept a set of images, a blackbox image classifier, and a saliency generator and will generate saliency maps for each image provided.\n",
+    "It will then combine the heatmaps of each corresponding class and plot them.\n",
+    "This should give us a view of the decision boundaries for each class similar to the one above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "08fb1a3e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from smqtk_classifier import ClassifyImage\n",
+    "from xaitk_saliency import GenerateImageClassifierBlackboxSaliency\n",
+    "from functools import reduce\n",
+    "from math import ceil\n",
+    "\n",
+    "def app(\n",
+    "    images: np.ndarray,\n",
+    "    image_classifier: ClassifyImage,\n",
+    "    saliency_generator: GenerateImageClassifierBlackboxSaliency\n",
+    "):\n",
+    "    num_imgs = images.shape[0]\n",
+    "    \n",
+    "    # Generate saliency maps\n",
+    "    sal_maps = []\n",
+    "    for i in range(num_imgs):    \n",
+    "        ref_image = images[i].reshape(28, 28)\n",
+    "        sal_maps.append(saliency_generator(ref_image, image_classifier))\n",
+    "    \n",
+    "    # Combine respective class maps\n",
+    "    combined_maps = reduce(np.add, sal_maps)\n",
+    "    \n",
+    "    # Plot combined maps\n",
+    "    plt.figure(figsize=(10, 5))\n",
+    "    for i in range(num_imgs):\n",
+    "        max_val = combined_maps[i].max()\n",
+    "        plt.subplot(2, ceil(num_imgs/2), i + 1)\n",
+    "        plt.imshow(combined_maps[i], cmap=plt.cm.RdBu, vmin=-max_val, vmax=max_val)\n",
```

Comment:
I'm not sure the use of `vmin=-max_val, vmax=max_val` is appropriate here. The value is not symmetric in the heatmaps. The scikit-learn example uses this because it takes the global coef matrix absolute-value max.

We also know via API documentation that saliency heatmaps are constrained to return in the `[-1,1]` range. We may want to normalize the `combined_maps` by `len(sal_maps)` and then pass `-1, 1` to `vmin, vmax` respectively. @brianhhu please double check me on this.

This generates the following output:
![image](https://user-images.githubusercontent.com/735270/134057813-1c56833a-f899-4f75-98e9-d176d8c722f5.png)

Additionally may want to output a color bar to be more clear on color meanings, unless we are intending to assume the scikit-learn example output is well understood.